src/go/rpk/pkg/cli/cluster/connections/list.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// Package connections deals with listing current connections in the cluster
+package connections
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	adminv2 "buf.build/gen/go/redpandadata/core/protocolbuffers/go/redpanda/core/admin/v2"
+	"connectrpc.com/connect"
+	"github.com/docker/go-units"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+const defaultPageSize = 32
+
+var errInvalidStateFilter = errors.New("invalid state filter")
+
+var stateOptionMap = map[string]adminv2.KafkaConnectionState{
+	"OPEN":   adminv2.KafkaConnectionState_KAFKA_CONNECTION_STATE_OPEN,
+	"CLOSED": adminv2.KafkaConnectionState_KAFKA_CONNECTION_STATE_CLOSED,
+}
+
+var tableHeaders = []string{
+	"NODE",
+	"SHARD",
+	"UID",
+	"CLIENT-ID",
+	"IP:PORT",
+	"OPEN-TIME",
+	"IDLE",
+	"PROD-TPUT",
+	"FETCH-TPUT",
+	"REQS/MIN",
+	"STATE",
+}
+
+func NewConnection(fs afero.Fs, p *config.Params) *cobra.Command {
```

Comment:
[nit] Can we move this into a connections.go file? basically to favor discoverability, is not clear why the root command of the package is in the list.go file.

Usually we split the root command and each command into its own file, that way the shared stuff can live on `connections.go` and the subcommands on its own.